overlord/snapstate/backend.go

```diff
@@ -83,6 +84,7 @@ type managerBackend interface {
 
 	// the undoers for install
 	UndoSetupSnap(s snap.PlaceInfo, typ snap.Type, installRecord *backend.InstallRecord, dev snap.Device, meter progress.Meter) error
+	CleanupKernelSnap(instanceName string, rev snap.Revision, meter progress.Meter) error
```

Comment:
why Cleanup and not UndoSetup as other methods here?

Reply:
I preferred to avoid the "Undo" bit as we use it in the handlers (undoSetupKernelSnap/undoCleanupOldKernelSnap), and if you do it to the last consequences you would end up with names starting like `undoUndoSetupKernelSnap` which is very confusing.

Reply:
Changed to use Remove